src/services/db/GitHubService.ts

```diff
@@ -791,7 +794,7 @@ export default class GitHubService {
   async changeRepoPrivacy(
     sessionData: { siteName: string; isomerUserId: string },
     shouldMakePrivate: boolean
-  ) {
+  ): Promise<ResultAsync<null, NotFoundError | GitHubApiError>> {
```

Comment:
nit: `ResultAsync` itself is a promise, so we don't need to wrap in a `Promise`

Reply:
In principle you are right but removing it here would cause error in typescript. See the attached screenshot. 
Because the ResultAsync type from the `neverthrow` library is a wrapper around a `Promise` that provides additional utility methods and properties. However, TypeScript doesn't recognize it as a global `Promise` because it's not an instance of the global `Promise` type.
<img width="1008" alt="Screenshot 2023-12-20 at 12 07 15 PM" src="https://github.com/isomerpages/isomercms-backend/assets/14878879/680811be-7304-4014-9c08-52f3cb902a1b">

Removing the `async` keyword would solve this error but we are using `await` inside the function, so it's not a viable option. 

Therefore, we should wrap `ResultAsync` inside the global `Promise` here for `Typescript` to correctly recognise it.

Reply:
Hmm I think it is a symptom of a bigger problem in the function where we are mixing `neverthrow` with async/await. Ideally, we should be wrapping the `this.axiosInstance.patch` inside a `ResultAsync.fromPromise`, then doing the error handling inside. That way, we can remove the async/await from this function and the return type is preserved as `ResultAsync`.

I'm ok with fixing this in a separate PR if you prefer.

Reply:
Yeah I think you are right. Refactored it in 1cb950bcd9aa3b41314912c8e30bb51cfc6e9c52
Wrapped the patch call inside `ResultAsync.fromPromise` as you suggested. 

Upon a quick search, there are other instances of similar mixed use of `async/await` and `ResultAsync`, will address them in separate PR.